tests/performance_tests/performance_test_basic.py

```diff
@@ -21,26 +21,144 @@
 relaunchTimeout = 30
 emptyBlockGoal = 5
 
-def waitForEmptyBlocks(node1):
+class blockData():
+    def __init__(self):
+        self.transactions = 0
+        self.cpu = 0
+        self.net = 0
+        self.elapsed = 0
+        self.time = 0
+        self.latency = 0
+
+class chainData():
+    def __init__(self):
+        self.blockLog = []
+    def sum_transactions(self):
+        total = 0
+        for block in self.blockLog:
+            total += block.transactions
+        return total
+    def sum_cpu(self):
+        total = 0
+        for block in self.blockLog:
+            total += block.cpu
+        return total
+    def sum_net(self):
+        total = 0
+        for block in self.blockLog:
+            total += block.net
+        return total
+    def sum_elapsed(self):
+        total = 0
+        for block in self.blockLog:
+            total += block.elapsed
+        return total
+    def sum_time(self):
+        total = 0
+        for block in self.blockLog:
+            total += block.time
+        return total
+    def sum_latency(self):
+        total = 0
+        for block in self.blockLog:
+            total += block.latency
+        return total
+    def print_stats(self):
+        print("Chain transactions: ", self.sum_transactions())
+        print("Chain cpu: ", self.sum_cpu())
+        print("Chain net: ", self.sum_net())
+        print("Chain elapsed: ", self.sum_elapsed())
+        print("Chain time: ", self.sum_time())
+        print("Chain latency: ", self.sum_latency())
+
+class chainsContainer():
+    def __init__(self):
+        self.preData = chainData()
+        self.totalData = chainData()
+        self.startBlock = 0
+        self.ceaseBlock = 0
+    def total_transactions(self):
```

Comment:
Reduce all the boilerplate methods into a single `total(self, attrname)` method which calls the built-in `getattr()` function and invoke it with each of the new cache attributes you'll be adding to `chainData`.